pkg/products/observability/reconciler.go

```diff
@@ -0,0 +1,369 @@
+				return phase, err
+			}
+		}
+		// If both namespaces are deleted, return complete
+		_, operatorNSErr := resources.GetNS(ctx, operatorNamespace, client)
+		_, productNSErr := resources.GetNS(ctx, productNamespace, client)
+		if k8serr.IsNotFound(operatorNSErr) && k8serr.IsNotFound(productNSErr) {
+			return integreatlyv1alpha1.PhaseCompleted, nil
+		}
+		return integreatlyv1alpha1.PhaseInProgress, nil
+	}, r.log)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, "Failed to reconcile finalizer", err)
+		return phase, err
+	}
+
+	phase, err = r.ReconcileNamespace(ctx, operatorNamespace, installation, client, r.log)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, fmt.Sprintf("Failed to reconcile %s ns", operatorNamespace), err)
+		return phase, err
+	}
+
+	phase, err = r.ReconcileNamespace(ctx, productNamespace, installation, client, r.log)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, fmt.Sprintf("Failed to reconcile %s ns", productNamespace), err)
+		return phase, err
+	}
+
+	phase, err = r.reconcileConfigMap(ctx, client)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, fmt.Sprintf("Failed to reconcile %s configmap which is required to disable observability operator initilisting it's own cr", ConfigMapNoInit), err)
+		return phase, err
+	}
+
+	phase, err = r.reconcileSubscription(ctx, client, installation, productNamespace, operatorNamespace)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, fmt.Sprintf("Failed to reconcile %s subscription", constants.ObservabilitySubscriptionName), err)
+		return phase, err
+	}
+
+	phase, err = r.reconcileComponents(ctx, client, productNamespace)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, "Failed to create components", err)
+		return phase, err
+	}
+
+	phase, err = monitoringcommon.ReconcileAlertManagerSecrets(ctx, client, r.installation, r.Config.GetNamespace(), alertManagerRouteName)
+	r.log.Infof("ReconcileAlertManagerConfigSecret", l.Fields{"phase": phase})
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		if err != nil {
+			r.log.Warning("failed to reconcile alert manager config secret " + err.Error())
+		}
+		events.HandleError(r.recorder, installation, phase, "Failed to reconcile alert manager config secret", err)
+		return phase, err
+	}
+
+	if string(r.Config.GetProductVersion()) != string(integreatlyv1alpha1.VersionObservability) {
+		r.Config.SetProductVersion(string(integreatlyv1alpha1.VersionObservability))
+		err := r.ConfigManager.WriteConfig(r.Config)
+		if err != nil {
+			return "", err
```

Comment:
So, we want to return an empty string instead of some kind of phase (`integreatlyv1alpha1.PhaseFailed`) here ? :thinking:

Reply:
Good catch, fixed in latest commit